client/src/pages/DatabaseView.jsx

```diff
@@ -1,84 +1,100 @@
+
 import React, { useState, useEffect } from 'react';
 import axios from 'axios';
 
 const DatabaseView = () => {
     const [data, setData] = useState([]);
-    const [dataType, setDataType] = useState('users'); // Default to 'users'
-    const [formData, setFormData] = useState({}); // For form data
-    const [isEditing, setIsEditing] = useState(false); // To toggle add/edit mode
-    const [editId, setEditId] = useState(null); // To store the id of the entity being edited
+    const [dataType, setDataType] = useState('users');
+    const [formData, setFormData] = useState({});
```

Comment:
the state for this shouldnt be an empty object since you expect things like username etc, either create an object with key value of the ones that you are using or split them out